gapic-generator-cloud/templates/cloud/client/method/def/_client.erb

```diff
@@ -22,12 +22,26 @@
 #
 def <%= method.name %> requests, options: nil, &block
   unless requests.is_a? Enumerable
-    raise ArgumentError, "requests must be an Enumerable"
+    if requests.respond_to? :to_enum
+      requests = requests.to_enum
+    else
+      raise ArgumentError, "requests must be an Enumerable"
+    end
   end
 
   requests = requests.lazy.map do |request|
+    # Google::Gax::Protobuf.coerce request, to: <%= method.request_type %>
     Google::Gax.to_proto request, <%= method.request_type %>
   end
 
-  <%= method.ivar %>.call(requests, options, &block)
+  # Converts hash and nil to an options object
+  options = Google::Gax::ApiCall::Options.new options.to_h if options.respond_to? :to_h
+  header_params = {} # { name: request.name }
```

Comment:
Still a WIP?

Reply:
This PR is not a WIP. It is not ever intended to be merged. It is for demonstration purposes only.